libsolidity/formal/SMTChecker.cpp

```diff
@@ -53,6 +54,25 @@ void SMTChecker::analyze(SourceUnit const& _source, shared_ptr<Scanner> const& _
 	m_scanner = _scanner;
 	if (_source.annotation().experimentalFeatures.count(ExperimentalFeature::SMTChecker))
 		_source.accept(*this);
+
+	solAssert(m_interface->solvers() > 0, "");
+	// If this check is true, Z3 and CVC4 are not available
+	// and the query answers were not provided, since SMTPortfolio
+	// guarantees that SmtLib2Interface is the first solver.
+	if (!m_interface->unhandledQueries().empty() && m_interface->solvers() == 1)
```

Comment:
Why do you need to check for `solvers == 1`?

Reply:
Because if there are more solvers this means either Z3 or CVC4 are available and we don't want to give this warning.

Reply:
Why would `unhandledQueries` be populated in that case?

Reply:
And `SMTPortfolio::unhandledQueries()` returns `SMTLib2Interface::unhandledQueries` because `CompilerStack` needs that

Reply:
So, the current behavior in such cases is: even if Z3/CVC4 are available, the unhandled queries are returned to CompilerStack anyway. We can of course change that to return nothing, and go back to unhandledQueries returning min(solvers | unhandledQueries)

Reply:
I suppose one could still submit the SMTLIB2 solutions and run again with 3 solvers if wanted.

Lets just keep the check.